### PR TITLE
Switch UTs to running on Dora code paths

### PR DESCRIPTION
**Pull request by JiamingMai**
Many tests do not apply or do not work under the new DORA architecture. They need to either be removed or fixed. This PR mainly adds the changes below:
1. Turn on DORA by default and make sure the UTs are actually testing against DORA (instead of the old code which we will not use any more)
2. Make the minimal set of config changes for tests to run and pass
3. For the impacted tests, skip them for now in order to unblock dev work (so we at least have some CI/CD). Those skipped tests are assigned to different people, who should later make the decision to whether to fix or remove them. There are simply a lot of tests, it's impossible to fix in this one PR.

Major config changes:
1) set `alluxio.dora.client.read.location.policy.enabled=true` by default
2) set `alluxio.master.worker.register.lease.enabled=false` by default
3) set `alluxio.user.netty.data.transmission.enabled=true` by default
4) set `alluxio.user.short.circuit.enabled=false` by default

Most of the skipped tests fail because:
1) Block Store is deprecated in DORA
2) Job master and job worker are deprecated in DORA
3) Master no longer manages metadata in DORA

The skipped tests are marked TODO in this format:
```
@Ignore
@DoraTestTodoItem(action = DoraTestTodoItem.Action.FIX, owner = "jiacheng",
    comment = "these tests should be moved to worker")
@Test
public void test()
```

### CODE REVIEW

**Comment by JiamingMai** [comment]
This PR mainly do the following things:
1) set alluxio.dora.client.read.location.policy.enabled=true by default
2) set alluxio.master.worker.register.lease.enabled=false by default
3) set alluxio.user.netty.data.transmission.enabled=true by default

These changes make many tests fail since the architecture of DORA is quite different from that of the previous Alluxio 2.9.x version. That is why this PR changed more than a hundred files. Most of the failed tests refer to the following reasons:
1) Block Store is deprecated in DORA
2) Job master and job worker are deprecated in DORA
3) Master no longer manages metadate in DORA

Though I have marked the depreacated tests with `@Ignore` to avoid failures, we still have to review these tests and check whether we need to fix them or not. This could be future work done in another PR.

For easier retrieval of these tests, I have left a `TODO` comment with a `<TEST_FIX>` string in the tests that are marked as `@Ignore`. We can review them later convienently.

**Comment by JiamingMai** [comment]
@Xenorith This test group takes too much time before testing leading to timeout error. However, actually all tests would be passed.

image

image

image

File: dora/core/client/fs/src/test/java/alluxio/client/block/stream/BlockInStreamTest.java [hunk 1]
```diff
@@ -90,6 +91,8 @@ public void before() throws Exception {
   }
 
   @Test
+  // TODO(JiamingMai): <TEST_FIX> Fix this test or remove it if it is deprecated
+  @Ignore("Block store is deprecated in DORA")
```
Comment (jiacheliu3) [inline dora/core/client/fs/src/test/java/alluxio/client/block/stream/BlockInStreamTest.java:95]: nit: can we @Ignore the whole class?

File: dora/core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 2]
```diff
@@ -3944,7 +3944,7 @@ public String toString() {
           .build();
   public static final PropertyKey MASTER_WORKER_REGISTER_LEASE_ENABLED =
       booleanBuilder(Name.MASTER_WORKER_REGISTER_LEASE_ENABLED)
-          .setDefaultValue(true)
+          .setDefaultValue(false)
```
Comment (jiacheliu3) [inline dora/core/common/src/main/java/alluxio/conf/PropertyKey.java:3947]: it's fine to disable it now, i think this leasing logic needs a bit of rework in dora

File: dora/core/common/src/main/java/alluxio/conf/PropertyKey.java [hunk 3]
```diff
@@ -7785,7 +7785,7 @@ public String toString() {
 
   public static final PropertyKey DORA_CLIENT_READ_LOCATION_POLICY_ENABLED =
       booleanBuilder(Name.DORA_CLIENT_READ_LOCATION_POLICY_ENABLED)
-          .setDefaultValue(false)
+          .setDefaultValue(true)
```
Comment (jiacheliu3) [inline dora/core/common/src/main/java/alluxio/conf/PropertyKey.java:7788]: i actually think we should separate this line of change to a separate PR like: "Enable Dora architecture by default" That's probably including removing key `WORKER_BLOCK_STORE_TYPE`

File: dora/integration/fuse/src/test/java/alluxio/fuse/AlluxioJnrFuseFileSystemTest.java [hunk 4]
```diff
@@ -78,6 +79,7 @@
  */
 @RunWith(PowerMockRunner.class)
 @PrepareForTest({BlockMasterClient.Factory.class})
+@Ignore("we don't maintain jnr any more")
```
Comment (jiacheliu3) [inline dora/integration/fuse/src/test/java/alluxio/fuse/AlluxioJnrFuseFileSystemTest.java:82]: good catch, we should create a separate PR to remove JNR

File: dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java [hunk 5]
```diff
@@ -30,7 +30,7 @@
  * The base class for all the shell integration test.
  */
 public abstract class AbstractShellIntegrationTest extends BaseIntegrationTest {
-  protected static final int SIZE_BYTES = Constants.MB * 16;
+  protected static final int SIZE_BYTES = Constants.MB * 1024;
```
Comment (jiacheliu3) [inline dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java:33]: relevant?

File: dora/tests/src/test/java/alluxio/client/cli/fs/command/HeadCommandIntegrationTest.java [hunk 6]
```diff
@@ -18,11 +18,14 @@
...
 
 import org.junit.Assert;
+import org.junit.Ignore;
 import org.junit.Test;
 
 /**
  * Tests for head command.
  */
+// TODO(JiamingMai): <TEST_FIX> Fix this test or remove it if it is deprecated
+@Ignore("Master no longer manages metadata")
```
Comment (jiacheliu3) [inline dora/tests/src/test/java/alluxio/client/cli/fs/command/HeadCommandIntegrationTest.java:28]: Honestly I don't know how to handle this kind of work. @Xenorith Is this related to your work on reorganizing commands? I figure we should find a way to keep the UTs, meaning it is fine to @Ignore them now but we should remember to dig them back up once the command line reorg is done.

**Review by beinan — APPROVED** [review]
lgtm

File: dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java (hunk 5 above)
Comment (JiamingMai) [reply dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java:33]: I found that some tests have to write large files that are larger than 16MB. This makes them failed. So I just set it larger to avoid failure.

File: dora/core/common/src/main/java/alluxio/conf/PropertyKey.java (hunk 3 above)
Comment (jiacheliu3) [reply dora/core/common/src/main/java/alluxio/conf/PropertyKey.java:7788]: Per discussion, this switch change itself cannot be achieved without the UT changes (because the UTs simply can't pass). So I give up and let's do it here.

File: dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java (hunk 5 above)
Comment (dbw9580) [reply dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java:33]: we should fix the failure of large files instead of making the failure go away

File: dora/core/client/fs/src/test/java/alluxio/client/block/stream/BlockInStreamTest.java (hunk 1 above)
Comment (huanghua78) [reply dora/core/client/fs/src/test/java/alluxio/client/block/stream/BlockInStreamTest.java:96]: I have the same question.

File: dora/integration/fuse/src/test/java/alluxio/fuse/AlluxioJnrFuseFileSystemTest.java (hunk 4 above)
Comment (huanghua78) [reply dora/integration/fuse/src/test/java/alluxio/fuse/AlluxioJnrFuseFileSystemTest.java]: @LuQQiu is planning to remove all JNR related code, right?

File: dora/tests/src/test/java/alluxio/client/cli/fs/command/CatCommandIntegrationTest.java [hunk 7]
```diff
@@ -34,6 +35,8 @@
 /**
  * Tests for cat command.
  */
+// TODO(JiamingMai): <TEST_FIX> Fix this test or remove it if it is deprecated
+@Ignore("Master no longer manages metadata")
 public final class CatCommandIntegrationTest extends AbstractFileSystemShellTest {
```
[the remark this answers is not in the document]
Comment (huanghua78) [reply dora/tests/src/test/java/alluxio/client/cli/fs/command/CatCommandIntegrationTest.java]: The sub test case `catAfterForceMasterSync` may not be applicable in Dora, but other test cases should still be valid.

File: dora/tests/src/test/java/alluxio/client/cli/fs/command/ChgrpCommandIntegrationTest.java [hunk 8]
```diff
@@ -18,13 +18,16 @@
...
+import org.junit.Ignore;
 import org.junit.Test;
 
 import java.io.IOException;
 
 /**
  * Tests for chgrp command.
  */
+// TODO(JiamingMai): <TEST_FIX> Fix this test or remove it if it is deprecated
+@Ignore("Master no longer manages metadata")
```
[the remark this answers is not in the document]
Comment (huanghua78) [reply dora/tests/src/test/java/alluxio/client/cli/fs/command/ChgrpCommandIntegrationTest.java]: The same to all `*CommandIntegrationTest`

[the remark this answers is not in the document]
Comment (jiacheliu3) [reply dora/tests/src/test/java/alluxio/client/cli/fs/command/ChgrpCommandIntegrationTest.java]: They don't work now, for this or that reason. You can have a try locally, and it'll be appreciated if you can push to this PR and fix a few :)

File: dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java (hunk 5 above)
Comment (jiacheliu3) [reply dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java:33]: I think instead of targeting to fix those tests in this change, we should merge this change in ASAP to unblock other changes. And track the disabled tests and strive to add them back with priority.

File: dora/core/common/src/main/java/alluxio/conf/PropertyKey.java (hunk 2 above)
Comment (elega) [reply dora/core/common/src/main/java/alluxio/conf/PropertyKey.java:3947]: well worker registration in 3.0 isn't as heavy-lifting as it used to be in 2.x so probably we have a chance to get rid of it completely

File: dora/core/common/src/main/java/alluxio/conf/PropertyKey.java (hunk 3 above)
Comment (elega) [reply dora/core/common/src/main/java/alluxio/conf/PropertyKey.java:7788]: should we rename this property? 
I feel this is the main switch of dora so maybe we should rename it to be alluxio.dora.enabled or somethign?

**Review by elega — APPROVED** [review]
lgtm thanks for the work

File: dora/core/common/src/main/java/alluxio/conf/PropertyKey.java (hunk 3 above)
Comment (jiacheliu3) [reply dora/core/common/src/main/java/alluxio/conf/PropertyKey.java:7788]: absolutely

Comment (jiacheliu3) [reply dora/core/common/src/main/java/alluxio/conf/PropertyKey.java:7788]: IMO it is fine whether we rename in a separate PR or do it here. If there's not much compatibility issue renaming it, then let's do it here.

File: dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java (hunk 5 above)
Comment (jiacheliu3) [reply dora/tests/src/test/java/alluxio/client/cli/fs/AbstractShellIntegrationTest.java:33]: @JiamingMai could you run again with `SIZE_BYTES = Constants.MB * 16` and paste the error here, so we know what we are hacking

**Comment by jiacheliu3** [comment]
> This PR mainly does the following things:
> 
> 1. set `alluxio.dora.client.read.location.policy.enabled=true` by default
> 2. set `alluxio.master.worker.register.lease.enabled=false` by default
> 3. set `alluxio.user.netty.data.transmission.enabled=true` by default
> 4. set `alluxio.user.short.circuit.enabled=false` by default
> 
> These changes make many tests fail since the architecture of DORA is quite different from that of the previous Alluxio 2.9.x version. That is why this PR changed more than a hundred files. Most of them failed caused by the following reasons:
> 
> 1. Block Store is deprecated in DORA
> 2. Job master and job worker are deprecated in DORA
> 3. Master no longer manages metadata in DORA
> 
> Though I have marked the deprecated tests with `@Ignore` to avoid failures, we still have to review these tests and check whether we need to fix them or not. This could be future work done in another PR.
> 
> For easier retrieval of these tests, I have left a `TODO` comment with a `<TEST_FIX>` string in the tests that are marked as `@Ignore`. We can review them later conveniently by searching the key word `<TEST_FIX>`.

Contents above is merged into the desc. And we invented a more structured way of tracking TODOs.

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please
